src/components/inputs/Counter/Counter.tsx

```diff
@@ -1,52 +1,71 @@
-import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
 
 import useLongPress from '../../../hooks/useLongPress/useLongPress';
+import { useRegistrationState } from '../../../state/contexts/RegistrationState';
 import './styles.css';
 
 function YearsCounter({
-  setCount,
-  count,
   maxValue,
   minValue,
   period = 'years',
 }: {
-  setCount: Dispatch<SetStateAction<number>>;
-  count: number;
   maxValue: number;
   minValue: number;
   period?: 'years' | 'days' | 'minutes';
 }) {
+  const [{ leaseDuration }, dispatchRegisterState] = useRegistrationState();
   const [registration, setRegistration] = useState('');
+  let initialCount = leaseDuration;
   const {
     handleOnClick: incHandleOnClick,
     handleOnMouseDown: incHandleOnMouseDown,
     handleOnMouseUp: incHandleOnMouseUp,
     handleOnTouchEnd: incHandleOnTouchEnd,
     handleOnTouchStart: incHandleOnTouchStart,
-  } = useLongPress(() => (count < maxValue ? setCount(++count) : null));
+  } = useLongPress(() =>
+    initialCount < maxValue
+      ? updateRegisterState({ key: 'setLeaseDuration', value: initialCount++ })
+      : null,
+  );
   const {
     handleOnClick: decHandleOnClick,
     handleOnMouseDown: decHandleOnMouseDown,
     handleOnMouseUp: decHandleOnMouseUp,
     handleOnTouchEnd: decHandleOnTouchEnd,
     handleOnTouchStart: decHandleOnTouchStart,
-  } = useLongPress(() => (count > minValue ? setCount(--count) : null));
+  } = useLongPress(() =>
+    initialCount > minValue
+      ? updateRegisterState({ key: 'setLeaseDuration', value: --initialCount })
+      : null,
+  );
 
   useEffect(() => {
     changePeriod();
-  }, [count]);
+  }, [leaseDuration]);
+
+  function updateRegisterState({ key, value }: { key: any; value: any }) {
+    // timeout to prevent jitter
+    setTimeout(
+      () =>
+        dispatchRegisterState({
+          type: key,
+          payload: value,
+        }),
+      50,
+    );
+  }
 
   function changePeriod() {
     const date = new Date();
     switch (period) {
       case 'years':
-        date.setFullYear(date.getFullYear() + count);
+        date.setFullYear(date.getFullYear() + leaseDuration);
         break;
       case 'days':
-        date.setDate(date.getDate() + count);
+        date.setDate(date.getDate() + leaseDuration);
         break;
       case 'minutes':
-        date.setHours(date.getHours() + count);
+        date.setHours(date.getHours() + leaseDuration);
```

Comment:
in an effort to keep the counter generic, keep `count`. i know you mentioned coming back to it in another PR, but at least at a `TODO` comment so we remember

Reply:
added todo